.github/workflows/SHA-version-update.yaml

```
@@ -0,0 +1,12 @@
+name: SHA version update
+
+on:
+  pull_request:
+    branches: [ master ]
+    paths: ['dependabot/pip/dot-github/dependabot/*']
+
+jobs:
+  update-sha:
+    runs-on: ubuntu-latest
+    steps:
```

```suggestion
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

You need to have the `actions/checkout@v2` step, otherwise the repository will not be checked out during the test, and as mentioned in my comment about the sparse checkouts, this should have fetch-depth set to 0.